cmd/mount_zfs/mount_zfs.c

```diff
@@ -50,6 +50,22 @@ libzfs_handle_t *g_zfs;
 static void
 parse_dataset(const char *target, char **dataset)
 {
+	/*
+	 * If a file or directory in your current working directory is
+	 * named 'dataset' then mount(8) will prepend your current working
+	 * directory to the dataset.  There is no way to prevent this
+	 * behavior so we simply check for it and strip the prepended
+	 * path when it is added.
+	 */
```

Comment:
Let's update this comment to mention an [an exception](https://github.com/karelzak/util-linux/commit/372ce5b74e79470b1bda1fc284c19a313a422361) was just added to mount for zfs as of util-linux 2.36.2.  But since it might be a while before this version rolls out everywhere it's needed let's keep the workaround in place for a while.